public/js/common.js

```diff
@@ -68,3 +68,113 @@ function failSnackbar(msg) {
 $(document).on('click', '#closeSnack', function() {
     $(this).parent().fadeOut();
 });
+
+
+/*
+These errors are only for UI use
+Error codes and their corresponding message. Error codes are under different
+categories:
+1000 -> system
+2000 -> user
+3000 -> question
+4000 -> class
+5000 -> analytics
+6000 -> import/export
+7000 -> settings
+*/
+
+const errors = Object.freeze({
+    //1000 system
+    1000: 'Invalid Request',
+    1001: 'Missing Requirement',
+    1002: 'Permission Denied',
+    1003: 'Confirm password doesn\'t match',
+    1004: 'Database connection failed',
+    1005: 'Database Error',
+    1006: 'Invalid username or password',
+    1007: 'A system error has occured',
+    1008: 'A system error has occured',
+    1009: 'A system error has occured',
+    1010: 'A system error has occured',
+    1011: 'A system error has occured',
+    1012: 'A system error has occured',
+    1013: 'A system error has occured',
+
+    //2000 user
+    2000: 'Invalid username or password',
+    2001: 'Failed to get student',
+    2002: 'Failed to get student list, please try again later!',
+    2003: 'Failed to get student list, please try again later!',
+    2004: 'Failed to get data',
+    2005: 'Failed to add admin',
+    2006: 'Failed to get user',
+    2007: 'Failed to add student',
+    2008: 'Failed to change student account status',
+    2009: 'User cannot be found',
+    2010: 'Password is incorrect',
+    2011: 'Failed to update profile',
+    2012: 'Failed to update',
+    2013: 'Failed to get users list',
+    2014: 'Failed to check if user exists',
+    2015: 'Failed to get student list, please try again later!',
+    2016: 'Failed to remove users',
+    2017: 'Failed to find user',
+    2018: 'Failed to update user',
+    2019: 'User already exists',
+    2020: 'Failed to get leaderboard',
```

Comment:
why is this user related

Reply:
because the points and the stats are related to the users